### PR TITLE
Fix: addon panic bug

### LINKED ISSUE #3015: vela addon enable velaux panic
我在自己搭建的gitlab上fork了catalog这个项目
使用这个地址enable velaux panic
image

### PR DESCRIPTION
**Pull request by wangyikewxgm**
Signed-off-by: wangyike <wangyike_wyk@163.com>

### Description of your changes

Fixes #3015 

I have:

- [ ] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: pkg/addon/source.go [hunk 1]
```diff
@@ -128,9 +128,13 @@ func NewAsyncReader(baseURL, bucket, subPath, token string, rdType ReaderType) (
...
 		u.Path = path.Join(u.Path, subPath)
 		tp, content, err := utils.Parse(u.String())
-		if err != nil || tp != utils.TypeGithub {
+		if err != nil {
 			return nil, err
 		}
+
+		if tp != utils.TypeGithub {
+			return nil, fmt.Errorf("%s registry haven't been supported yet", tp)
+		}
```
Comment (wonderflow) [inline pkg/addon/source.go:137]: can you write a test about the panic case?

**Review by zzhzero — COMMENTED** [review]
1. 如果未支持的类型属于一个错误那是不是在parse里面直接返回err好点
2. tp为local,oss都是现在未被支持的类型么?
3. unknown registry haven't been supported yet 这个错误提示是不是有点怪

**Comment by wangyikewxgm** [comment]
> 1. 如果未支持的类型属于一个错误那是不是在parse里面直接返回err好点
> 2. tp为local,oss都是现在未被支持的类型么?
> 3. unknown registry haven't been supported yet 这个错误提示是不是有点怪

愿意提个pr解一下这个问题吗？：）

**Comment by wangyikewxgm** [comment]
> 1. 如果未支持的类型属于一个错误那是不是在parse里面直接返回err好点
> 2. tp为local,oss都是现在未被支持的类型么?
> 3. unknown registry haven't been supported yet 这个错误提示是不是有点怪

Will you submit a pr to solve this probelm?

File: pkg/addon/source.go (hunk 1 above)
Comment (wangyikewxgm) [reply pkg/addon/source.go]: fixed

**Review by wonderflow — APPROVED** [review]

**Comment by wonderflow** [comment]
> 1. 如果未支持的类型属于一个错误那是不是在parse里面直接返回err好点
> 2. tp为local,oss都是现在未被支持的类型么?
> 3. unknown registry haven't been supported yet 这个错误提示是不是有点怪

均已修改，感谢review

**Pull request merged**